### PR TITLE
Sso basic auth

### PR DESCRIPTION
**Pull request by NakulManchanda**

### CODE REVIEW

**Comment by NakulManchanda** [comment]
@jackjia-ibm @jordanCain @stevenhorsman 
Please review PR for sso changes,
Did rebase the branch to staging as well

There is one problem which is leading to failure of integration tests
curl -k --request GET --url "${host_name}/${dataset_endpoint}/NAKUL" --header "authorization: Bearer ${apimlAuthToken}"

Where NAKUL is search string

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/AbstractZosmfDataSetService.java [hunk 1]
```diff
@@ -9,8 +9,6 @@
  */
```
Comment (jordanCain) [inline data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/AbstractZosmfDataSetService.java:9]: Make sure all copyright headers are updated where necessary

File: data-sets-api-server/src/main/java/org/zowe/data/sets/controller/AbstractDataSetsController.java [hunk 2]
```diff
@@ -41,45 +38,40 @@
 
 import java.net.URI;
 
-@RestController
-@RequestMapping("/api/v1/datasets")
-@Api(value = "Data Sets APIs", tags = "Data Sets APIs")
-public class DataSetsController {
+public abstract class AbstractDataSetsController extends AbstractApiController {
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/data/sets/controller/AbstractDataSetsController.java]: to limit path to 
api/v1/datasets

api/v1/unifiles dont have that path
https://github.com/zowe/data-sets/pull/156/commits/e4adf02de325b68ea295889bd66de9836f69a284

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/AbstractZosmfDataSetService.java (hunk 1 above)
Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/AbstractZosmfDataSetService.java:9]: updated

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetService1.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (jordanCain) [inline data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetService1.java:1]: For consistency this file should be name ZosmfDataSetService**V**1.java

and in line the v2 one should be renamed to include **V**

File: data-sets-api-server/src/main/java/org/zowe/unix/files/services/zosmf/ZosmfUnixFilesService1.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (jordanCain) [inline data-sets-api-server/src/main/java/org/zowe/unix/files/services/zosmf/ZosmfUnixFilesService1.java:1]: Again rename this to include **V**

Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/unix/files/services/zosmf/ZosmfUnixFilesService1.java]: done

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetService1.java (hunk 3 above)
Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetService1.java]: done

**Comment by NakulManchanda** [comment]
@jordanCain 
- updated readme based on jobs readme
- refactor missing "V" from service class names

**Pull request merged**